lib/Sema/TypeCheckType.cpp

```diff
@@ -2148,6 +2148,13 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
   if (!ty) ty = resolveType(repr, instanceOptions);
   if (!ty || ty->hasError()) return ty;
 
+  // Return if the type is a DependentMemberType and we're in a structural
+  // resolution stage
```

Comment:
The code below is clear enough that this comment adds no new info. However, an explanation _why_ we're doing this would be nice (in the structural stage, protocol type aliases are not yet resolved, so we don't want to error out on a member type that will later become a function type).

Reply:
I have fixed the comment!